src/lib/edge-functions/proxy.cjs

```diff
@@ -88,26 +88,24 @@ const initializeProxy = async ({
     port: isolatePort,
     projectDir,
   })
-  const hasEdgeFunctions = userFunctionsPath !== undefined || internalFunctions.length !== 0
 
   return async (req) => {
+    const registry = await server
+    if (!registry) return
+    const hasEdgeFunctions = registry.functions.length !== 0
```

Comment:
The current approach prevents us from initialising the Deno server unnecessarily when edge functions aren't being used. Initialising the Deno server involves potentially downloading the Deno CLI, which is a big deal. We'll lose that with your change.

Rather than checking that we have declarations for internal functions, could we perhaps check that the internal functions directory exists, and proceed with initialising the server if so? That way, if someone is not using edge functions, there's no reason for that directory to exist and therefore we can keep the optimisation for those users.

Does that make sense?

Reply:
Yeah this makes sense!